x/fbridge/module/abci.go

```diff
@@ -0,0 +1,70 @@
+package module
+
+import (
+	"fmt"
+	sdk "github.com/Finschia/finschia-sdk/types"
+	"github.com/Finschia/finschia-sdk/x/fbridge/keeper"
+	"github.com/Finschia/finschia-sdk/x/fbridge/types"
+)
+
+func EndBlocker(ctx sdk.Context, k keeper.Keeper) {
+	guardianTrustLevel := k.GetParams(ctx).GuardianTrustLevel
+	proposals := k.GetRoleProposals(ctx)
+	for _, proposal := range proposals {
+		if ctx.BlockTime().After(proposal.ExpiredAt) {
```

Comment:
How about moving the pruning into `BeginBlocker()`?. Then, in `DeliverTx` there would be no possibility of accessing expired proposals.